### ISSUE TITLE
Pop Shell stopped working because of the version of GNOME

### ISSUE THREAD

**Issue by open-antux** [pop-os/shell#1216]
**(1) Issue/Bug Description:**
After a system upgrade, pop shell stopped working because "The extension is not compatible with the current version of GNOME".

**(2) Steps to reproduce (if you know):**
Upgrade GNOME to 40.4

**(3) Expected behavior:**

**(4) Distribution (run `cat /etc/os-release`):**
 ```bash
$ cat /etc/os-release 
NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo Linux"
ANSI_COLOR="1;32"
HOME_URL="https://www.gentoo.org/"
SUPPORT_URL="https://www.gentoo.org/support/"
BUG_REPORT_URL="https://bugs.gentoo.org/"
VERSION_ID="2.8"
```

**(5) Gnome Shell version:**
```bash
$ gnome-shell --version
GNOME Shell 41.0
```

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**
```bash
$ emerge --info gnome-extra/gnome-shell-extension-pop-shell
Portage 3.0.28 (python 3.9.7-final-0, default/linux/amd64/17.1/desktop/gnome/systemd, gcc-11.2.0, glibc-2.33-r7, 5.14.10-gentoo x86_64)
=================================================================
                         System Settings
=================================================================
System uname: Linux-5.14.10-gentoo-x86_64-Intel-R-_Core-TM-_i5-8600K_CPU_@_3.60GHz-with-glibc2.33
KiB Mem:    16331188 total,  11211080 free
KiB Swap:    2097148 total,   2097148 free
Timestamp of repository gentoo: Mon, 18 Oct 2021 08:00:01 +0000
Head commit of repository gentoo: 059b020a5d84fca3290ca2b2becf9c334df29073
Timestamp of repository brother-overlay: Sun, 19 Sep 2021 21:51:57 +0000
Head commit of repository brother-overlay: 09ee0c1e201a47821384e215b1f4e60daa6c2e9c

sh bash 5.1_p8
ld GNU ld (Gentoo 2.37_p1 p0) 2.37
app-shells/bash:          5.1_p8::gentoo
dev-java/java-config:     2.3.1::gentoo
dev-lang/perl:            5.34.0-r3::gentoo
dev-lang/python:          3.9.7_p1::gentoo, 3.10.0_p1::gentoo
dev-lang/rust:            1.55.0::gentoo
dev-util/cmake:           3.21.3::gentoo
sys-apps/baselayout:      2.8::gentoo
sys-apps/sandbox:         2.25::gentoo
sys-devel/autoconf:       2.13-r1::gentoo, 2.71-r1::gentoo
sys-devel/automake:       1.16.5::gentoo
sys-devel/binutils:       2.37_p1::gentoo
sys-devel/gcc:            11.2.0::gentoo
sys-devel/gcc-config:     2.4::gentoo
sys-devel/libtool:        2.4.6-r6::gentoo
sys-devel/make:           4.3::gentoo
sys-kernel/linux-headers: 5.14::gentoo (virtual/os-headers)
sys-libs/glibc:           2.33-r7::gentoo
Repositories:

gentoo
    location: /var/db/repos/gentoo
    sync-type: rsync
    sync-uri: rsync://rsync.gentoo.org/gentoo-portage
    priority: -1000
    sync-rsync-verify-jobs: 1
    sync-rsync-extra-opts: 
    sync-rsync-verify-max-age: 24
    sync-rsync-verify-metamanifest: yes

brother-overlay
    location: /var/db/repos/brother-overlay
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/brother-overlay.git
    masters: gentoo

deepin
    location: /var/lib/layman/deepin
    sync-type: laymansync
    sync-uri: git://github.com/zhtengw/deepin-overlay.git
    masters: gentoo
    priority: 50

fordfrog
    location: /var/lib/layman/fordfrog
    sync-type: laymansync
    sync-uri: https://anongit.gentoo.org/git/dev/fordfrog.git
    masters: gentoo
    priority: 50

ACCEPT_KEYWORDS="amd64 ~amd64"
ACCEPT_LICENSE="*"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-march=native -O2 -pipe"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/share/config /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo /etc/texmf/language.dat.d /etc/texmf/language.def.d /etc/texmf/updmap.d /etc/texmf/web2c"
CXXFLAGS="-march=native -O2 -pipe"
DISTDIR="/var/cache/distfiles"
EMERGE_DEFAULT_OPTS="--jobs=6 --load-average=6"
ENV_UNSET="CARGO_HOME DBUS_SESSION_BUS_ADDRESS DISPLAY GOBIN GOPATH PERL5LIB PERL5OPT PERLPREFIX PERL_CORE PERL_MB_OPT PERL_MM_OPT XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR"
FCFLAGS="-march=native -O2 -pipe"
FEATURES="assume-digests binpkg-docompress binpkg-dostrip binpkg-logs config-protect-if-modified distlocks ebuild-locks fixlafiles ipc-sandbox merge-sync multilib-strict network-sandbox news parallel-fetch pid-sandbox preserve-libs protect-owned qa-unresolved-soname-deps sandbox sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-march=native -O2 -pipe"
GENTOO_MIRRORS="https://gentoo.mirror.garr.it/ http://gentoo.mirror.garr.it/"
LANG="it_IT.utf8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
MAKEOPTS="-j6 -l6"
PKGDIR="/var/cache/binpkgs"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/tmp"
USE="X a52 aac acl acpi alsa amd64 bluetooth branding bzip2 cairo cdda cdr cli colord crypt cups dbus dri dts dvd dvdr eds emboss encode evo exif flac fortran gdbm gentoo-vm gif gnome gnome-keyring gnome-online-accounts gpm gstreamer gtk gui iconv icu introspection ipv6 jpeg lcms libglvnd libnotify libsecret libtirpc mad mng mp3 mp4 mpeg multilib nautilus ncurses networkmanager nls nptl nvidia ogg opengl openmp pam pango pcre pdf png policykit ppds pulseaudio readline sdl seccomp spell split-usr ssl startup-notification svg sysprof systemd tcpd tiff tracker truetype udev udisks unicode upower usb vorbis wayland wxwidgets x264 xattr xcb xml xv xvid zlib" ABI_X86="64" ADA_TARGET="gnat_2019" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="karbon sheets words" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="mmx mmxext sse sse2" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx" GRUB_PLATFORMS="efi-64" INPUT_DEVICES="libinput" KERNEL="linux" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" LUA_SINGLE_TARGET="lua5-1" LUA_TARGETS="lua5-1" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php7-3 php7-4" POSTGRES_TARGETS="postgres12 postgres13" PYTHON_SINGLE_TARGET="python3_9" PYTHON_TARGETS="python3_9" RUBY_TARGETS="ruby30" USERLAND="GNU" VIDEO_CARDS="nvidia" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq proto steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, INSTALL_MASK, LC_ALL, LINGUAS, PORTAGE_BINHOST, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS, RUSTFLAGS

=================================================================
                        Package Settings
=================================================================

gnome-extra/gnome-shell-extension-pop-shell-1.2.0_p20210727::gentoo was built with the following:
USE="" ABI_X86="(64)"
```

**(7) Where was Pop Shell installed from:**
Installed by [gnome-extra/gnome-shell-extension-pop-shell](https://packages.gentoo.org/packages/gnome-extra/gnome-shell-extension-pop-shell) package

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**
1x 1080p

**(9) Other Installed/Enabled Extensions:**
Blur my shell

**(10) Other Notes:**

**Comment by Bananaman** [comment]
```
gsettings set org.gnome.shell disable-extension-version-validation "true"
```

Turn off GNOME extension version checks, until this repo adds 41 to the list.

**Comment by Bananaman** [comment]
It has been added:

https://github.com/pop-os/shell/commit/d39f185ee16cb024f86163ea2f7d24c7e15b695d

**Comment by Alch-Emi** [comment]
Should @djelic's contribution be turned into a PR?  That seems like an important change, and with the release of Fedora 35, (and i'm sure many other distros) there's going to be more and more people on Gnome 41, and having the extension claim support but cause crashes is probably not a good thing

**Comment by Tedium16** [comment]
I have it working on Arch Linux on GNOME 41, but I stopped using it cause, it wasn't causing crashes but caused certain windows to appear stretched with sections cut of the screen when it was activated.

**Comment by Alch-Emi** [comment]
I'm getting that too on F35, even with djelic's patch.  Maybe that should be a different issue though.  Honestly didn't know it was caused by popshell until your post :p

**Comment by jacobgkau** [comment]
> caused certain windows to appear stretched with sections cut of the screen when it was activated.



> Maybe that should be a different issue though.

https://github.com/pop-os/shell/issues/1232 is open for this.

**Comment by jmmaranan** [comment]
> Had the same issue @open-antux mentioned, fixed with following patch. Hope it helps!
> 
> ```
> diff --git a/src/extension.ts b/src/extension.ts
> index 90038c8..70ea04e 100644
> --- a/src/extension.ts
> +++ b/src/extension.ts
> @@ -2638,7 +2638,7 @@ let default_getcaption_workspace: any;
>   */
>  function _show_skip_taskbar_windows(ext: Ext) {
>      let cfg = ext.conf;
> -    if (!GNOME_VERSION?.startsWith("40.")) {
> +    if (!(GNOME_VERSION?.startsWith("40.") || GNOME_VERSION?.startsWith("41."))) {
>          // TODO GNOME 40 added a call to windowtracker and app var is not checked if null
>          // in WindowPreview._init(). Then new WindowPreview() is being called on
>          // _addWindowClone() of workspace.js.
> @@ -2810,7 +2810,7 @@ function _show_skip_taskbar_windows(ext: Ext) {
>   *
>   */
>  function _hide_skip_taskbar_windows() {
> -    if (!GNOME_VERSION?.startsWith("40.")) {
> +    if (!(GNOME_VERSION?.startsWith("40.") || GNOME_VERSION?.startsWith("41."))) {
>          if (default_isoverviewwindow_ws)
>              Workspace.prototype._isOverviewWindow = default_isoverviewwindow_ws;
>      }
> ```

The above version check is applied generically on #1250 and version check would be removed due to #1255 if it is merged.
Though this should already be closed by #1221 and all other reported issues here seems to be tracked already.

**Comment by jacobgkau** [comment]
Closing this because it was fixed by https://github.com/pop-os/shell/pull/1221. We'll make sure https://github.com/pop-os/shell/pull/1255 doesn't cause this to regress.
